src/Tools/dotnet-monitor/Extensibility/ProgramExtension.cs

```diff
@@ -75,20 +76,42 @@ public async Task<EgressArtifactResult> EgressArtifact(ExtensionEgressPayload co
             // This is really weird, yes, but this is one of 2 overloads for [Stream].WriteAsync(...) that supports a CancellationToken, so we use a ReadOnlyMemory<char> instead of a string.
             ReadOnlyMemory<char> NewLine = new ReadOnlyMemory<char>("\r\n".ToCharArray());
 
-            string exeName = Declaration.Program;
+            ProcessStartInfo pStart = new ProcessStartInfo()
+            {
+                RedirectStandardInput = true,
+                RedirectStandardOutput = true,
+                RedirectStandardError = true,
+                CreateNoWindow = true,
+                UseShellExecute = false,
+            };
 
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            IFileInfo executableInfo;
+            if (Declaration.UseSharedDotNetHost)
             {
-                exeName += ".exe";
-            }
+                // .NET Monitor currently only runs with the shared framework, thus its ProcessPath
+                // will be the shared dotnet host (dotnet.exe).
+                executableInfo = new PhysicalFileInfo(new FileInfo(Environment.ProcessPath));
```

Comment:
For images, won't this result int the /app/dotnet-monitor path?

Reply:
You're probably right. I'll have to think about how to differentiate between the apphost and the dotnet host from managed code.

Reply:
The dotnet root resolution code is at https://github.com/dotnet/runtime/blob/7392fb7c4617ee831d160f0c61aaf43038a6aba8/src/native/corehost/fxr_resolver.cpp#L55

Order of looking this up is:
1. Check if hostfxr library is in the app directory, if so, assume self-contained.
1. Check for DOTNET_ROOT(_<arch>) environment variable
1. On Windows, check for installation in registry; on *nix, check `/etc/dotnet` for path file
1. Check default install location

We can probably get away with not doing this ourselves but instead doing:
1. If current process is named `dotnet`, just use that one.
1. Get location of currently loaded hostfxr library; if in app directory, then assume self-contained
1. If DOTNET_ROOT(_<arch>) is set, use that path.
1. Get dotnet relative to the hostfxr library path.

.NET Monitor doesn't ship self-contained at this time, so we don't need to handle that scenario just yet.

Reply:
I mostly adopted the apphost's lookup algorithm (the first list).

Reply:
I was able to test this on Windows and Ubuntu on WSL2. The Windows code path fell back to the default installation whereas the Linux code path used the self-registered location.